src/Agent/NewRelic/Agent/Core/Wrapper/AgentWrapperApi/Data/MethodCallData.cs

```diff
@@ -8,12 +8,14 @@ public class MethodCallData
         public readonly string TypeName;
         public readonly string MethodName;
         public readonly int InvocationTargetHashCode;
+        public readonly bool IsAsync;
 
-        public MethodCallData(string typeName, string methodName, int invocationTargetHashCode)
+        public MethodCallData(string typeName, string methodName, int invocationTargetHashCode, bool isAsync = false)
```

Comment:
Would it be hard to make the isAsync parameter required? I find that it is easier to track down cases where `isAsync` is `false` when the parameter is required.

Reply:
That was my first thought, too, but there are 133 places in unit tests that create a `MethodCallData` that would need updating. The good news is that there are only two places in Agent code that create a `MethodCallData`. The primary one takes a `MethodCall`, and it's required in that structure.

Reply:
We could use .ctor chaining to handle that case if you think it would be helpful.
```
public MethodCallData(string typeName, string methodName, int invocationTargetHashCode) : this(typeName, methodName, invocationTargetHashCode, false) {}

public MethodCallData(string typeName, string methodName, int invocationTargetHashCode, bool isAsync)
{
   //Initialize the type here
}
```

Reply:
Ah, I see what you mean now. I thought you were concerned with catching all the correct uses on this initial implementation, but you want to be able to find the synchronous cases more easily. Neither of the two (non-testing) places where the `MethodCallData` constructor is called use the default param, so I think I'll leave it as-is.